jwst/ami/tests/test_ami_interface.py

```diff
@@ -7,28 +7,29 @@
 from jwst.ami import AmiAnalyzeStep
 
 
-def test_ami_analyze_cube_fail():
-    """Make sure ami_analyze fails if input is CubeModel (_calints)"""
+@pytest.fixture()
+def example_model():
     model = datamodels.CubeModel((25, 19, 19))
     model.meta.instrument.name = "NIRISS"
     model.meta.instrument.filter = "F277W"
     model.meta.observation.date = "2019-01-01"
     model.meta.observation.time = "00:00:00"
-    with pytest.raises(RuntimeError):
-        AmiAnalyzeStep.call(model)
+    return model
 
 
-def test_ami_analyze_no_reffile_fail(monkeypatch):
+@pytest.mark.parametrize("oversample", [2, 4])
+def test_ami_analyze_even_oversample_fail(example_model, oversample):
+    """Make sure ami_analyze fails if oversample is even"""
+    with pytest.raises(ValueError, match="Oversample value must be an odd integer."):
+        AmiAnalyzeStep.call(example_model, oversample=oversample)
+
+
+def test_ami_analyze_no_reffile_fail(monkeypatch, example_model):
     """Make sure that ami_analyze fails if no throughput reffile is available"""
-    model = datamodels.ImageModel((19, 19))
-    model.meta.instrument.name = "NIRISS"
-    model.meta.instrument.filter = "F277W"
-    model.meta.observation.date = "2019-01-01"
-    model.meta.observation.time = "00:00:00"
 
     def mockreturn(input_model, reftype, observatory=None, asn_exptypes=None):
         return "N/A"
     monkeypatch.setattr(stpipe.crds_client, 'get_reference_file', mockreturn)
 
-    with pytest.raises(RuntimeError):
-        AmiAnalyzeStep.call(model)
+    with pytest.raises(RuntimeError, match="No throughput reference file found."):
```

Comment:
This test will need to be updated and possibly expanded into several tests for the different reference files.